src/cpp/rtps/builtin/data/ParticipantProxyData.cpp

```diff
@@ -36,63 +36,70 @@ namespace eprosima {
+    , m_VendorId(c_VendorId_Unknown)
+    , m_expectsInlineQos(false)
+    , m_availableBuiltinEndpoints(0)
+    , metatraffic_locators(allocation.locators.max_unicast_locators, allocation.locators.max_multicast_locators)
+    , default_locators(allocation.locators.max_unicast_locators, allocation.locators.max_multicast_locators)
+    , m_manualLivelinessCount(0)
 #if HAVE_SECURITY
-    security_attributes_(0UL),
-    plugin_security_attributes_(0UL),
+    , security_attributes_(0UL)
+    , plugin_security_attributes_(0UL)
 #endif
-    isAlive(false),
-    mp_leaseDurationTimer(nullptr)
+    , isAlive(false)
+    , lease_duration_event(nullptr)
+    , should_check_lease_duration(false)
+    , m_readers(allocation.readers)
+    , m_writers(allocation.writers)
     {
     }
 
-ParticipantProxyData::ParticipantProxyData(const ParticipantProxyData& pdata) :
-    m_protocolVersion(pdata.m_protocolVersion),
-    m_guid(pdata.m_guid),
-    m_VendorId(pdata.m_VendorId),
-    m_expectsInlineQos(pdata.m_expectsInlineQos),
-    m_availableBuiltinEndpoints(pdata.m_availableBuiltinEndpoints),
-    m_metatrafficUnicastLocatorList(pdata.m_metatrafficUnicastLocatorList),
-    m_metatrafficMulticastLocatorList(pdata.m_metatrafficMulticastLocatorList),
-    m_defaultUnicastLocatorList(pdata.m_defaultUnicastLocatorList),
-    m_defaultMulticastLocatorList(pdata.m_defaultMulticastLocatorList),
-    m_manualLivelinessCount(pdata.m_manualLivelinessCount),
-    m_participantName(pdata.m_participantName),
-    m_key(pdata.m_key),
-    m_leaseDuration(pdata.m_leaseDuration),
+ParticipantProxyData::ParticipantProxyData(const ParticipantProxyData& pdata) 
+    : m_protocolVersion(pdata.m_protocolVersion)
+    , m_guid(pdata.m_guid)
+    , m_VendorId(pdata.m_VendorId)
+    , m_expectsInlineQos(pdata.m_expectsInlineQos)
+    , m_availableBuiltinEndpoints(pdata.m_availableBuiltinEndpoints)
+    , metatraffic_locators(pdata.metatraffic_locators)
+    , default_locators(pdata.default_locators)
+    , m_manualLivelinessCount(pdata.m_manualLivelinessCount)
+    , m_participantName(pdata.m_participantName)
+    , m_key(pdata.m_key)
+    , m_leaseDuration(pdata.m_leaseDuration)
 #if HAVE_SECURITY
-    identity_token_(pdata.identity_token_),
-    permissions_token_(pdata.permissions_token_),
-    security_attributes_(pdata.security_attributes_),
-    plugin_security_attributes_(pdata.plugin_security_attributes_),
+    , identity_token_(pdata.identity_token_)
+    , permissions_token_(pdata.permissions_token_)
+    , security_attributes_(pdata.security_attributes_)
+    , plugin_security_attributes_(pdata.plugin_security_attributes_)
 #endif
-    isAlive(pdata.isAlive),
-    m_properties(pdata.m_properties),
-    m_userData(pdata.m_userData),
-    mp_leaseDurationTimer(nullptr)
+    , isAlive(pdata.isAlive)
+    , m_properties(pdata.m_properties)
+    , m_userData(pdata.m_userData)
+    , lease_duration_event(nullptr)
+    , should_check_lease_duration(false)
```

Comment:
We have to study if `m_readers` and `m_writers` attributes have to be copied too.

Reply:
This method is only called from SecurityManager when a new participant is discovered and the corresponding DiscoveredParticipantInfo struct is created. Only participant info is used, so there is no need to copy m_readers and m_writers

Reply:
Added comments on commit 8500a0c